From 0db9228c15489da4dcac1c0fcf60a1f49fb58c33 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Fri, 12 Sep 2014 11:48:46 -0400
Subject: [ethernet] i40e: fix panic due to too-early Tx queue enable

Message-id: <1410522528-19793-9-git-send-email-sassmann@redhat.com>
Patchwork-id: 93940
O-Subject: [RHEL7.1 PATCH 08/10] i40e: fix panic due to too-early Tx queue enable
Bugzilla: 1140994
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

This fixes the panic under traffic load when resetting.  This issue
could also show up if/whenever there is a Tx-timeout.

Change-ID: Ie393a1f17fd5d962e56fc3bfe784899ef25402f5
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 32b5b81170b6a60cf41d403ab31c417b56ff0d44)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 56d432f..44c5562 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -5289,7 +5289,7 @@ static void i40e_fdir_reinit_subtask(struct i40e_pf *pf)
  **/
 static void i40e_vsi_link_event(struct i40e_vsi *vsi, bool link_up)
 {
- if (!vsi)
+ if (!vsi || test_bit(__I40E_DOWN, &vsi->state))
   return;
 
  switch (vsi->type) {
-- 
1.7.1